pkg/cli/version/version.go

```diff
@@ -1,25 +1,184 @@
+	return &VersionOptions{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdVersion is copied from upstream NewCmdVersion with addition of OpenShift Server version info.
+// OpenShift Server version is output only if logged in to a cluster as an admin user.
+func NewCmdVersion(fullName string, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewVersionOptions(ioStreams)
+	ko := kversion.NewVersionOptions(ioStreams)
 	cmd := &cobra.Command{
-		Use:   "version",
-		Short: "Display version",
-		Long:  "Display version",
+		Use:     "version",
+		Short:   i18n.T("Print the client and server version information"),
+		Long:    "Print the client and server version information for the current context",
+		Example: fmt.Sprintf(versionExample, fullName),
 		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Fprintf(out, "%s %v\n", fullName, versionInfo)
+			cmdutil.CheckErr(ko.Validate())
+			cmdutil.CheckErr(o.Complete(f, cmd, ko))
+			cmdutil.CheckErr(o.Run(ko))
 		},
 	}
-
+	cmd.Flags().BoolVar(&ko.ClientOnly, "client", ko.ClientOnly, "Client version only (no server required).")
+	cmd.Flags().BoolVar(&ko.Short, "short", ko.Short, "Print just the version number.")
+	cmd.Flags().StringVarP(&ko.Output, "output", "o", ko.Output, "One of 'yaml' or 'json'.")
 	return cmd
 }
+
+// Complete is copied from upstream version command with added clusteroperator client
+// to report OpenShift server version
+func (o *VersionOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, ko *kversion.VersionOptions) error {
+	var err error
+	if ko.ClientOnly {
+		return nil
+	}
+	o.discoveryClient, err = f.ToDiscoveryClient()
+	// if we had an empty rest.Config, continue and just print out client information.
+	// if we had an error other than being unable to build a rest.Config, fail.
+	if err != nil && !clientcmd.IsEmptyConfig(err) {
+		return err
+	}
+
+	clientConfig, err := f.ToRESTConfig()
+	if err != nil && !clientcmd.IsEmptyConfig(err) {
+		return err
+	}
+	if clientConfig != nil {
+		o.oClient, err = configv1client.NewForConfig(clientConfig)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Run is copied from upstream version command, with added OpenShift server version logic
+func (o *VersionOptions) Run(ko *kversion.VersionOptions) error {
+	var versionInfo Version
+	clientVersion := version.Get()
+	versionInfo.ClientVersion = &clientVersion
+
+	var serverErr error
+	var serverVersion *apimachineryversion.Info
+	if !ko.ClientOnly {
+		if o.discoveryClient != nil {
+			// Always request fresh data from the server
+			o.discoveryClient.Invalidate()
+			serverVersion, serverErr = o.discoveryClient.ServerVersion()
+			versionInfo.ServerVersion = serverVersion
+		}
+		if o.oClient != nil {
+			var clusterOperator *configv1.ClusterOperator
+			clusterOperator, serverErr = o.oClient.ClusterOperators().Get("openshift-apiserver", metav1.GetOptions{})
+			// if error here, log and move on
```

Comment:
I don't see a log line

Reply:
I did have a log there, decided against it.  Without it, output of cmd is exactly the same as 3.x when not logged in as admin.  With the log statement, output differs.  I'll fix the comment, though.

Reply:
I've added the log back in, for klog.V(5)